File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int counter = 0;
    int answer = 0;
    int kthSmallest(TreeNode* root, int k) {
        traverse(root, k);
        return answer;
    }
    void traverse(TreeNode*& root, int k){
        if(root == nullptr){
            return;
        }
        traverse(root->left, k);
        counter++;
        if(counter == k){
            answer = root->val;
            return;
        }
        traverse(root->right, k);

    
    }
};

File: Data Structures & Algorithms/binary-tree-postorder-traversal/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
        List<Integer> answer = new ArrayList<>();
        post(root, answer);
        return answer;
    }
    private void post(TreeNode root, List<Integer> answer){
        if(root == null){
            return;
        }
        post(root.left, answer);
        post(root.right, answer);
        answer.add(root.val);
    }
}